api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/EriDSDCohortQueries.java

```diff
@@ -1380,182 +1400,222 @@ public CohortDefinition getBreastfeedingPatientsStartingART() {
+   * );
+   *
+   * return cd; }
+   */
 
-    return cd;
-  }*/
+  /**
+   * Get Patients who are pregnant and also breastfeeding
+   *
+   * @return
+   */
+  /*
+   * public CohortDefinition
+   * getPatientsWhoArePregnantAndNotBreastFeedingAndParticipateInDsdModelUnstable(
+   * ) { CompositionCohortDefinition cd = new CompositionCohortDefinition();
+   *
+   * cd.setName("N: Pregnant: includes all pregnant patients");
+   * cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+   * cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+   * cd.addParameter(new Parameter("location", "Location", Location.class));
+   *
+   * cd.addSearch( "pregnant", EptsReportUtils.map(
+   * getPatientsWhoArePregnantInLast9Months(),
+   * "startDate=${startDate},endDate=${endDate},location=${location}"));
+   * cd.addSearch( "breastfeeding", EptsReportUtils.map(
+   * getPatientsWhoAreBreastfeedingInLast18Months(),
+   * "startDate=${startDate},endDate=${endDate},location=${location}"));
+   * cd.addSearch( "activeAndUnstable", EptsReportUtils.map(
+   * getPatientsWhoAreActiveAndParticipateInDsdModelUnstable(),
+   * "startDate=${startDate},endDate=${endDate},location=${location}"));
+   *
+   * cd.setCompositionString("activeAndUnstable AND (pregnant OR breastfeeding)");
+   *
+   * return cd; }
+   */
 
   /**
-   * Get patients who are breastfeeding and not pregnant
+   * Get All patients that are participating in DSD Model
    *
    * @return
    */
-  /* public CohortDefinition
-      getPatientsWhoAreBreastFeedingAndNotPregnantAndParticipateInDsdModelUnstable() {
+  /*
+   * private CohortDefinition getAllPatientsParticipatingInDsdModel() {
+   * SqlCohortDefinition cd = new SqlCohortDefinition();
+   *
+   * cd.setName("All Patients participating in >=1 measured DSD model");
+   * cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+   * cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+   * cd.addParameter(new Parameter("location", "Location", Location.class));
+   *
+   * cd.setQuery( DsdQueries.getPatientsParticipatingInDsdModel(
+   * hivMetadata.getPrevencaoPositivaInicialEncounterType().getEncounterTypeId(),
+   * hivMetadata.getPrevencaoPositivaSeguimentoEncounterType().getEncounterTypeId(
+   * ), hivMetadata.getGaac().getConceptId(),
+   * hivMetadata.getFamilyApproach().getConceptId(),
+   * hivMetadata.getAccessionClubs().getConceptId(),
+   * hivMetadata.getSingleStop().getConceptId(),
+   * hivMetadata.getRapidFlow().getConceptId(),
+   * hivMetadata.getQuarterlyDispensation().getConceptId(),
+   * hivMetadata.getCommunityDispensation().getConceptId(),
+   * hivMetadata.getAnotherModel().getConceptId(),
+   * hivMetadata.getStartDrugs().getConceptId(),
+   * hivMetadata.getContinueRegimen().getConceptId())); return cd; }
+   */
+
+  private CohortDefinition getBreastfeedingExcludingPregnantPatients() {
```

Comment:
The breastfeeding cohort definition should be re-used from TX_NEW. Better if we extract it to a separate class maybe `HivCohortQueries`.

Reply:
done